### ISSUE TITLE
SEC-2172: Login Page redirect Issue with GAE

### ISSUE THREAD

**Issue by spring-projects-issues** [spring-projects/spring-security#2399]
[Joey Wilkinson](https://jira.spring.io/secure/ViewProfile.jspa?name=monkeymanjojo) (Migrated from [SEC-2172](https://jira.spring.io/browse/SEC-2172?redirect=false)) said:

I have been working on implementing Spring Security into my GWT web application, but having issues with basic Spring Security login redirects.  The redirects work when I don't have GAE enabled in eclipse(in project settings), but when I enable GAE the redirect to the login page stops working.

I get the same issue when setup a very basic spring security on top of the default GWT GAE sample project when you click new GWT project in eclipse(with all of the spring libraries):

Here is what my http looks like:

<http auto-config="true">
        <intercept-url pattern="/**" access="ROLE_USER" />
        <form-login/>
</http>

Basically I'm trying to make my whole gwt application required to be logged in with access="ROLE_USER" with a redirect to Spring Security's default login page for users to login.

Please help
Please help, I've been told Spring Security works well with GWT

Has anyone experienced this kind of issue before?  Or try to replicate issue?
